packages/babel-helper-hoist-variables/src/index.ts

```diff
@@ -38,8 +38,14 @@ const visitor = {
     > = path.get("declarations");
     let firstId;
 
+    const needsRename =
+      path.node.kind !== "var" && path.parentPath.isBlockStatement();
+
     for (const declar of declarations) {
       firstId = declar.node.id;
+      if (needsRename) {
+        declar.scope.rename(declar.node.id.name);
```

Comment:
The TS warning here is correct; I think this breaks, for example, with `const { x } = {}`. You probably need to do
```js
if (needsRename) {
  for (const name of Object.keys(declar.getBindingIdentifiers()) {
    declar.scope.rename(name);
  }
}
```

Reply:
Okay, I will do that.

Reply:
Also add a test for this!

Reply:
Okay, I will add the test

Reply:
I think that it's working now.